### PR TITLE
[WIP] Fixing minor issue mentioned in #342 with the deeplab ar detector

### PR DESCRIPTION
**Pull request by elbashandy**

### CODE REVIEW

**Comment by elbashandy** [comment]
@burlen I am using shallow_copy to use for the cartesian mesh output. To delete the 'IVT' variable, should I do a full copy and then remove the variable?

The deletion is request number 3 in #342

File: apps/teca_deeplabv3p_ar_detect.in [hunk 1]
```diff
@@ -26,8 +26,8 @@ def main():
 
     parser.add_argument(
         '--output_file', type=str, required=False,
-        default=r'pytorch_deeplabv3p_ar_detect_%t%.nc',
-        help=r'file pattern for output netcdf files (%t% is the time index)'
+        default=r'deeplabv3p_ar_detect_%%t%%.nc',
```
Comment (burlen) [inline apps/teca_deeplabv3p_ar_detect.in:29]: % in the default value needs to be left alone, in the help string it needs to be escaped. the way to escape it is with %.

**Comment by burlen** [comment]
> @burlen I am using shallow_copy to use for the cartesian mesh output. To delete the 'IVT' variable, should I do a full copy and then remove the variable?

```
mesh_out->get_point_data()->remove("IVT")
```
A shallow copy makes a new pointer to the original data, removing that pointer is safe to do.

**Comment by burlen** [comment]
@taobrienlbl & @elbashandy - I looked at the build logs and failures in the latest builds seem to be quota related. 
```
 Server Response:
 Warning: fopen(/home/user/cdash/backup/TECA_Travis_CI_ubuntu_18-04_streaming_mode_47c1787_20200403-2320-Continuous_158595606636.44_Configure.xml.lock): failed to open stream: Disk quota exceeded in /home/user/cdash/include/ctestparser.php
```
Not sure if that's the same missing file issue Travis reports above. This one is a server side issue, ie quota limit is at nersc, and therefor it should be reported to nersc. I also had some issues in my latest push. Results were not posted to cdash and it is the same issue.

@elbashandy would you please file a ticket with nersc?

**Comment by elbashandy** [comment]
@burlen Sure will take care of it

**Comment by elbashandy** [comment]
@burlen is it possible to add nested teca_metadata types in Python? when I do something like this it returns an error:

```
ar_probability_atts = teca_py.teca_metadata()
ar_probability_atts["long_name"] = "posterior AR flag"
ar_probability_atts["units"] = "probability"

out_md = out_mesh.get_metadata()
attributes = out_md["attributes"]
attributes["ar_probability"] = ar_probability_atts
```

The error:
```
Traceback (most recent call last):
  File "/app/TECA_build/lib/teca_py.py", line 8839, in __getitem__
    return _teca_py.teca_metadata___getitem__(self, name)
TypeError: in method 'teca_metadata___getitem__', argument 2 of type 'std::string const &'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/TECA_build/lib/teca_deeplabv3p_ar_detect.py", line 484, in execute
    attributes["ar_probability"] = ar_probability_atts
  File "/app/TECA_build/lib/teca_py.py", line 8827, in __setitem__
    return _teca_py.teca_metadata___setitem__(self, name, value)
SystemError: <built-in function teca_metadata___setitem__> returned a result with an error set
ERROR: [0:140189082703168] [/project/TECA/io/teca_cf_writer.cxx:709 3.0.0-16-g27d0905]
ERROR: input mesh 0 is empty input or not a cartesian mesh
```

My guess is that it only accepts types that can be added to a `teca_variant_array` as implemented in swig:
```
/* md['name'] = value */
    PyObject *__setitem__(const std::string &name, PyObject *value)
    {
        teca_py_gil_state gil;

        Py_INCREF(Py_None);

        p_teca_variant_array varr;
        if ((varr = teca_py_object::new_variant_array(value))
            || (varr = teca_py_array::new_variant_array(value))
            || (varr = teca_py_sequence::new_variant_array(value))
            || (varr = teca_py_iterator::new_variant_array(value)))
        {
            self->set(name, varr);
            return Py_None;
        }

        TECA_PY_ERROR(PyExc_TypeError,
            "Failed to convert value for key \"" << name << "\"")

        Py_DECREF(Py_None);
        return nullptr;
    }
```

Please let me know if I am missing something.

**Comment by burlen** [comment]
> @burlen is it possible to add nested teca_metadata types in Python?

Yes, this was an omission in the interface. I'll add it to my streaming_mode  PR, since the new parallel I/O capabilities require all arrays to have to have well defined attributes in the metadata. It would be OK w/ me if you get your other fixes in now, so long as we don't forget to circle back to this when the interface is fixed.

**Comment by burlen** [comment]
@elbashandy updates to Python are in the commit I just pushed here. you should be good to go. the py_test_metadata has an example of setting/getting nested metadata. Also, note that the latest develop branch has 2 commits that I made to get the deeplab tests to pass on my system. You'll want to rebase from develop to get those, before delving into the test failures here. They look like different issues to me.

**Comment by taobrienlbl** [comment]
Looks like this covers everything mentioned in #342 -- nice, and thanks @elbashandy and @burlen !

**Comment by elbashandy** [comment]
Thanks @burlen!

**Pull request merged**